### ISSUE TITLE
Air keeps hanging and not reloading under wsl2

### ISSUE THREAD

**Issue by salirzv** [air-verse/air#679]
I have a standard air config generated by init command. problem is that sometime air just stops responding and not watching the files anymore and I cant even exit out of it by Ctrl + C and I need to reboot wsl to make it quit.

I have no idea where to begin to look for error since the error is so random and general. any ideas?

everything is the latest version, WSL is a standard debian instance

![Screenshot 2024-11-05 192304](https://github.com/user-attachments/assets/e5bdef95-1809-4626-8271-2823694d852b)

**Comment by istyf** [comment]
You seem to have an error on line 9 in your own main.tmpl that causes the run to stop. You have to start by fixing that issue. 

I have seen similar lockups on my WSL2 machine and hope that #670 and #674 gets reviewed soon to fix those issues.

**Comment by salirzv** [comment]
> You seem to have an error on line 9 in your own main.tmpl that causes the run to stop. You have to start by fixing that issue.
> 
> I have seen similar lockups on my WSL2 machine and hope that #670 and #674 get reviewed soon to fix those issues.

@istyf I'm aware of the error, other projects that I used go with air the app kept rebuilding after an error on file change.
and it happens even if there is no error

**Comment by istyf** [comment]
I agree. It should work in either case. Let’s hope that someone has the time to look at the PR:s soon.

**Comment by gmetme** [comment]
I'm seeing the same thing

**Comment by nikolasmorado** [comment]
I don't think this is exclusive to WSL, I run arch and get the same issue on both my laptop and desktop. From what I can tell it happens when running air for a very long time.

**Comment by krtfbn** [comment]
Unresponsive to `Ctrl+c`, `Ctrl+d` and file changes when I run air in WSL2 with `.wslconfig`
```
[wsl2]
processors=1
```
Fixed with changing `processors=2` or commenting out this line.
